test/unit/CMakeLists.txt

```diff
@@ -39,6 +39,10 @@ FOREACH (source_file ${check_PROGRAMS})
     TARGET_LINK_LIBRARIES (
         ${source_file_we} mqtt_cpp_iface Boost::unit_test_framework
     )
+    IF (MSVC AND MQTT_USE_STATIC_OPENSSL)
+        TARGET_LINK_LIBRARIES (${source_file_we} Crypt32)
+    ENDIF ()
+
```

Comment:
Why this update is required?

Reply:
Ah yes. This is again maybe unrelated. But openssl static om Windows requires linking crypt32

Reply:
https://stackoverflow.com/questions/37522654/linking-with-openssl-lib-statically

Reply:
Could you remove this?
static link is out of scope here.

Reply:
I noticed that I already merged static support. Hmm...
Maybe it is required. What does 32 mean?

Reply:
It seems that 32bit. It this appropriate on 64bit environment.

I personally hate this kind of things. So I choose header only library. But there are many kind of things in CMakeLists.txt.
Sigh...

Reply:
> It seems that 32bit. It this appropriate on 64bit environment.

I mean "Is Crypt32 the appropriate choice on both 32 bit and 64 bit environment ?"
If it is, then I use this part.

Reply:
Yes, it is strange. But I think it is also for 64 builds
https://docs.microsoft.com/en-us/windows/win32/seccrypto/crypt32-dll-versions

Reply:
Yes, the dependencies of openssl are annoying. Also the zlib needs to be compiled static. So maybe only provide dynamic openssl on windows.

Reply:
Maybe you can look at it like this: it is responsibility of openssl to link against crypt32, not this project. So hopefully the static builds of openssl on windows will be fixed in the future.

Reply:
Here is issue of openssl:
https://github.com/openssl/openssl/issues/1061

Reply:
https://github.com/openssl/openssl/blob/master/NOTES-Windows.txt#L167-L172

Reply:
Or cmake ? 
https://gitlab.kitware.com/cmake/cmake/-/issues/19263

Reply:
Unfortunately, I don't have much time to investigate it.
So if 

```cmake
    IF (MSVC AND MQTT_USE_STATIC_OPENSSL)
        TARGET_LINK_LIBRARIES (${source_file_we} Crypt32)
    ENDIF ()
```

is required for CI, then I apply it.

Otherwise, if this works well for both 64bit and 32bit, then I apply it.
If it works either 32bit or 64bit only, I don't apply it.

Which is the status ?

Reply:
It works for both 64bit and 32bit

Reply:
It is not needed for CI, because CI builds with dynamic openssl.

Reply:
> It works for both 64bit and 32bit

Ok, I accept it.